sample-controller/controller.go

```diff
@@ -37,32 +38,33 @@ func main() {
 	// Ignoring some errors for brevity
 	cfg, _ := clientcmd.BuildConfigFromFlags("", "")
 	kubeClient, _ := kubernetes.NewForConfig(cfg)
-
+	ctx := context.Background()
 	// Get a pod informer
 	kubeInformerFactory := kubeinformers.NewSharedInformerFactory(kubeClient, time.Second*30)
 	informer := kubeInformerFactory.Core().V1().Pods().Informer()
 
 	loggingHandler := &LoggerHandler{
-		DefaultHandler: kubehandler.DefaultHandler{
+		DefaultHandler: kubehandlerv2.DefaultHandler{
 			Informer: informer,
 			Synced:   informer.HasSynced,
 		},
 	}
 
 	// This name is used as the workqueue name
-	loop := kubehandler.NewEventLoop("logger_queue")
+	loop := kubehandlerv2.NewEventLoop("logger_queue")
 
 	// Register all your handlers
 	loop.Register(loggingHandler)
 
 	// We're not handling signals for clean teardown. For production code, you
 	// probably want to do that
+	// We could also pass the context's done channel as well.
 	stopCh := make(chan struct{})
```

Comment:
If we are not using this, this can be removed

Reply:
Have removed the usage of stopCh and started using context